main/src/cgeo/geocaching/models/Geocache.java

```diff
@@ -639,6 +639,12 @@ public String getGeocode() {
         return geocode;
     }
 
+    @NonNull
+    public String getShortGeocode() {
+        final int maxLength = (geocode.length() < 8) ? geocode.length() : 8;
+        return geocode.substring(0, maxLength);
```

Comment:
IMHO we should add the "…" character, if the Geocode got shortened. What do you think?

Reply:
Something like...

```
return (geocode.length() <= 8) ? geocode : (geocode.substring(0, 8) + "…");
```

Reply:
I would leave it as it is, but it could be done of course.

Reply:
In this special case for ALC its relatively irrelevant for the user to know, that the geocode is shortened. But as AFAICS this is a generic implementation we should IMHO include the ellipsis character.

Reply:
I wouldn't mind adding it if that suits the overall style better. After all, the code sequence @fm-sys  provided is much cooler than what I came up with.

Reply:
I am now using @fm-sys' statement,  however without the dots but those could now be added with a much simpler modification later.

Reply:
> Something like...
> 
> ```
> @NonNull
> public String getShortGeocode() {
>     return (geocode.length() <= 8) ? geocode : (geocode.substring(0, 8) + "…");
> }
> ```
> 
> Or even better, exclude the "8" to a private static class variable

@fm-sys : The "8" is needed in more than one class now. What is the best place to define it as a constant such that it get's included everywhere and can be referenced where needed, and changed only once when needed?

Reply:
I would place it in Geocache.java

But also the shortening should be done in static utility function (also in Geocache.java) so that if we want to change the design of short geocode only one place needs to be edited

Reply:
I also need it in `Waypoint.java` and in `GeoitemRef.java`. I had hoped there could be a system wide constant, perhaps somewhere like: `main/res/values/numbers.xml`

Reply:
As I wrote, you should **not** rawly use it in  `Waypoint.java` or `GeoitemRef.java`! Create a utility function instead...

```
private static final int SHORT_GEOCODE_MAX_LENGTH = 8;

...

@NonNull
public String getShortGeocode() {
    return generateShortGeocode(geocode);
}


@NonNull
public String generateShortGeocode(final String fullGeocode) {
    return (fullGeocode.length() <= 8) ? fullGeocode: (fullGeocode.substring(0, 8) + "…");
}
```

So that `generateShortGeocode` can be used also in `Waypoint.java` and `GeoitemRef.java`. 

The general rule should be: Avoid duplications. Ideally, when you want to change something in the future, you should only have to edit one single place, cause no one will remember such hidden code duplications.

Reply:
agreed, but shouldn't `generateShortCode` then be some static helper method in `utils/Formatter`, e. g. `formatShortItemCode()`?

Reply:
Yes, you are right, that's even better.  :)